spec/lib/alaveteli_geoip_spec.rb

```diff
@@ -0,0 +1,123 @@
+# -*- encoding : utf-8 -*-
+require File.expand_path(File.dirname(__FILE__) + '/../spec_helper')
+
+describe AlaveteliGeoIP do
+
+	describe '.country_code_from_ip' do
+
+    it 'delegates to an instance of the class' do
+      instance = double
+      allow(AlaveteliGeoIP).to receive(:instance).and_return(instance)
+      allow(instance).to receive(:country_code_from_ip).with('127.0.0.1')
+      AlaveteliGeoIP.country_code_from_ip '127.0.0.1'
+    end
+
+	end
+
+  describe '.instance' do
+
+    it 'creates a new instance' do
+      expect(AlaveteliGeoIP.instance).to be_instance_of(AlaveteliGeoIP)
+    end
+
+    it 'caches the instance' do
+      expect(AlaveteliGeoIP.instance).to equal(AlaveteliGeoIP.instance)
+    end
+
+  end
+
+  describe '.new' do
+
+    it 'configures the instance with the configured country code' do
+      expect(AlaveteliGeoIP.new.current_code).
+        to eq(AlaveteliConfiguration::iso_country_code)
+    end
+
+    describe 'if there is a geoip database configured and present' do
+
+      it 'configures the instance with an instance of geoip' do
+        allow(File).to receive(:file?).
+          with(AlaveteliConfiguration::geoip_database).and_return(true)
+        expect(GeoIP).to receive(:new).with(AlaveteliConfiguration::geoip_database)
+        AlaveteliGeoIP.new.geoip
+      end
+
+    end
+
+    describe 'if there is only a Gaze URL configured' do
+
+      it 'configures the instance with the Gaze URL' do
+        allow(AlaveteliConfiguration).to receive(:geoip_database).and_return(nil)
+        expect(AlaveteliGeoIP.new.gaze_url).
+         to equal(AlaveteliConfiguration::gaze_url)
```

Comment:
Wonder if we should explicitly stub the gaze_url (as `http://gaze.example.net`) and then assert `eq('http://gaze.example.net')`?

This should be `eq` rather than `equal`, I think, because `equal` is for object equality, and `'a'.equal? 'a' # => false`